system/lib/pthread/library_pthread.c

```diff
@@ -159,6 +159,20 @@ static void init_em_queued_call_args(em_queued_call* q,
   }
 }
 
+static em_queued_call* em_queued_call_create(EM_FUNC_SIGNATURE sig,
+                                             void* func,
+                                             void* satellite,
+                                             va_list args) {
+  em_queued_call* call = em_queued_call_malloc();
+  if (call) {
```

Comment:
How about just `assert(call)` here instead and we can remove the two `assert(x)` that exist at the two call sites?   Fewer asserts and fewer conditionals overall?

Reply:
I've generally been trying to handle allocation failures correctly in this file since all the new functions can correctly return errors in that case. Do you think we should be assuming that allocations are always successful instead?

Reply:
As far as I know handling malloc failures is not something we try to do anywhere in the codebase.   In general in emscripten malloc failures will abort so even the assert is maybe overkill (@kripken correct me if I'm wrong).

Reply:
Actually I guess we do give folks the ability to turn off ABORTING_MALLOC.. so maybe its reasonable check malloc return values, but there are certainly a lot of places where we don't.   I've been working on threading code a lot recently we certainly don't check any of the allocations made thread startup..  we should probably have a consistent approach here.

Reply:
Certainly +1 for a consistent approach, and I would be happy to remove the checks after `malloc` if everyone's on board with that.

Reply:
I think we have a TODO to do this in `library_pthread.c`, e.g.
```c
  em_queued_call* q = em_queued_call_malloc();
  assert(q);
  // TODO: handle errors in a better way, this pattern appears in several places
  //       in this file. The current behavior makes the calling thread hang as
  //       it waits (for synchronous calls).
  // If we failed to allocate, return 0 which means we did not execute anything
  // (we also never will in that case).
  if (!q)
    return 0;
```
+1 for a consistent approach, although there are some non-obvious tradeoffs here so I'm not sure offhand what to suggest.